pkg/graph/resolvers/trb_request_document_test.go

```diff
@@ -0,0 +1,116 @@
+	fileToUpload := bytes.NewReader([]byte("Test file content"))
+	gqlInput := model.CreateTRBRequestDocumentInput{
+		RequestID:            documentToCreate.TRBRequestID,
+		DocumentType:         documentToCreate.CommonDocumentType,
+		OtherTypeDescription: &documentToCreate.OtherType,
+		FileData: graphql.Upload{
+			File:        fileToUpload,
+			Filename:    documentToCreate.FileName,
+			Size:        fileToUpload.Size(),
+			ContentType: "application/pdf",
+		},
+	}
+
+	createdDocument, err := CreateTRBRequestDocument(
+		suite.testConfigs.Context,
+		suite.testConfigs.Store,
+		suite.testConfigs.S3Client,
+		gqlInput,
+	)
+	suite.NoError(err)
+	suite.NotNil(createdDocument)
+
+	checkDocumentEquality(suite, documentToCreate, suite.testConfigs.Principal.ID(), trbRequestID, createdDocument)
+	suite.EqualValues(suite.testConfigs.S3Client.GetBucket(), createdDocument.Bucket)
+
+	return createdDocument // used by other tests
+}
+
+func getTRBRequestDocumentsByRequestIDSubtest(suite *ResolverSuite, trbRequestID uuid.UUID, createdDocument *models.TRBRequestDocument) {
+	documents, err := GetTRBRequestDocumentsByRequestID(
+		suite.testConfigs.Context,
+		suite.testConfigs.Store,
+		suite.testConfigs.S3Client,
+		trbRequestID,
+	)
+	suite.NoError(err)
+	suite.Equal(1, len(documents))
+
+	fetchedDocument := documents[0]
+	suite.NotNil(fetchedDocument)
+
+	checkDocumentEquality(suite, createdDocument, createdDocument.CreatedBy, createdDocument.TRBRequestID, fetchedDocument)
+}
+
+func deleteTRBRequestDocumentSubtest(suite *ResolverSuite, createdDocument *models.TRBRequestDocument) {
+	deletedDocument, err := DeleteTRBRequestDocument(suite.testConfigs.Context, suite.testConfigs.Store, createdDocument.ID)
+	suite.NoError(err)
+	checkDocumentEquality(suite, createdDocument, createdDocument.CreatedBy, createdDocument.TRBRequestID, deletedDocument)
+
+	remainingDocuments, err := GetTRBRequestDocumentsByRequestID(
+		suite.testConfigs.Context,
+		suite.testConfigs.Store,
+		suite.testConfigs.S3Client,
+		createdDocument.TRBRequestID,
+	)
+	suite.NoError(err)
+	suite.Equal(0, len(remainingDocuments))
+}
+
+func checkDocumentEquality(
+	suite *ResolverSuite,
+	expectedDocument *models.TRBRequestDocument,
+	expectedCreatedBy string,
+	expectedTRBRequestID uuid.UUID,
+	actualDocument *models.TRBRequestDocument,
+) {
+	// baseStruct fields
+	suite.NotNil(actualDocument.ID)
+	suite.EqualValues(expectedCreatedBy, actualDocument.CreatedBy)
+	suite.NotNil(actualDocument.CreatedAt)
+	suite.Nil(actualDocument.ModifiedBy)
+	suite.Nil(actualDocument.ModifiedAt)
+
+	// TRBRequestDocument-specific fields
+	suite.EqualValues(expectedTRBRequestID, actualDocument.TRBRequestID)
+	suite.EqualValues(expectedDocument.CommonDocumentType, actualDocument.CommonDocumentType)
+	suite.EqualValues(expectedDocument.OtherType, actualDocument.OtherType)
+	suite.EqualValues(expectedDocument.FileName, actualDocument.FileName)
```

Comment:
Could be nice to fetch the document that was uploaded and check the [sha](https://pkg.go.dev/crypto/sha256) matches that of the file which was sent.

Reply:
I thought about it, but ultimately that's at least partly testing the behavior of MinIO, so it's not really giving us confidence that the code works as desired in a deployed environment.